File: dbflow/src/androidTest/java/com/raizlabs/android/dbflow/test/structure/foreignkey/ForeignKeyTest.java
package com.raizlabs.android.dbflow.test.structure.foreignkey;

import com.raizlabs.android.dbflow.sql.language.Select;
import com.raizlabs.android.dbflow.test.FlowTestCase;
import com.raizlabs.android.dbflow.test.structure.autoincrement.TestModelAI;

import static com.raizlabs.android.dbflow.test.structure.foreignkey.ForeignModel_Table.name;

public class ForeignKeyTest extends FlowTestCase {

    // region Test Foreign Key

    public void testForeignKey() {
        ForeignParentModel parentModel = new ForeignParentModel();
        parentModel.setName("Test");
        parentModel.save();

        ForeignModel foreignModel = new ForeignModel();
        foreignModel.testModel1 = parentModel;
        foreignModel.setName("Test");
        foreignModel.save();

        ForeignModel retrieved = new Select().from(ForeignModel.class)
                .where(name.is("Test"))
                .querySingle();
        assertNotNull(retrieved);
        assertNotNull(retrieved.testModel1);
        assertEquals(retrieved.testModel1, foreignModel.testModel1);
    }

    public void testForeignKey2() {

        TestModelAI testModelAI = new TestModelAI();
        testModelAI.name = "TestAI";
        testModelAI.save();

        ForeignModel2 foreignModel2 = new ForeignModel2();
        foreignModel2.testModelAI = testModelAI;
        foreignModel2.setName("Test");
        foreignModel2.save();

        ForeignModel2 retrieved = new Select().from(ForeignModel2.class)
                .where(ForeignModel2_Table.name.is("Test"))
                .querySingle();
        assertNotNull(retrieved);
        assertNotNull(retrieved.testModelAI);
        assertEquals(retrieved.testModelAI, foreignModel2.testModelAI);
    }

    // endregion Test Foreign Key
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/structure/foreignkey/ForeignKeyTest.kt
package com.raizlabs.android.dbflow.structure.foreignkey

import com.raizlabs.android.dbflow.sql.language.Select
import com.raizlabs.android.dbflow.FlowTestCase
import com.raizlabs.android.dbflow.structure.autoincrement.TestModelAI

import org.junit.Test

import com.raizlabs.android.dbflow.structure.foreignkey.ForeignModel_Table.name
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull

class ForeignKeyTest : FlowTestCase() {

    // region Test Foreign Key

    @Test
    fun testForeignKey() {
        val parentModel = ForeignParentModel()
        parentModel.name = "Test"
        parentModel.save()

        val foreignModel = ForeignModel()
        foreignModel.testModel1 = parentModel
        foreignModel.name = "Test"
        foreignModel.save()

        val retrieved = Select().from(ForeignModel::class.java)
                .where(name.`is`("Test"))
                .querySingle()
        assertNotNull(retrieved)
        assertNotNull(retrieved!!.testModel1)
        assertEquals(retrieved.testModel1, foreignModel.testModel1)
    }

    @Test
    fun testForeignKey2() {

        val testModelAI = TestModelAI()
        testModelAI.name = "TestAI"
        testModelAI.save()

        val foreignModel2 = ForeignModel2()
        foreignModel2.testModelAI = testModelAI
        foreignModel2.name = "Test"
        foreignModel2.save()

        val retrieved = Select().from(ForeignModel2::class.java)
                .where(ForeignModel2_Table.name.`is`("Test"))
                .querySingle()
        assertNotNull(retrieved)
        assertNotNull(retrieved!!.testModelAI)
        assertEquals(retrieved.testModelAI, foreignModel2.testModelAI)
    }

    // endregion Test Foreign Key
}
